jib-core/src/main/java/com/google/cloud/tools/jib/frontend/JavaLayerConfigurations.java

```diff
@@ -36,47 +37,61 @@
   /** Represents the different types of layers for a Java application. */
   @VisibleForTesting
   enum LayerType {
-    DEPENDENCIES("dependencies", JavaEntrypointConstructor.DEFAULT_DEPENDENCIES_PATH_ON_IMAGE),
+    DEPENDENCIES(
+        "dependencies", JavaEntrypointConstructor.DEFAULT_DEPENDENCIES_PATH_ON_IMAGE, true),
     SNAPSHOT_DEPENDENCIES(
-        "snapshot dependencies", JavaEntrypointConstructor.DEFAULT_DEPENDENCIES_PATH_ON_IMAGE),
-    RESOURCES("resources", JavaEntrypointConstructor.DEFAULT_RESOURCES_PATH_ON_IMAGE),
-    CLASSES("classes", JavaEntrypointConstructor.DEFAULT_CLASSES_PATH_ON_IMAGE),
+        "snapshot dependencies",
+        JavaEntrypointConstructor.DEFAULT_DEPENDENCIES_PATH_ON_IMAGE,
+        true),
+    RESOURCES("resources", JavaEntrypointConstructor.DEFAULT_RESOURCES_PATH_ON_IMAGE, true),
+    CLASSES("classes", JavaEntrypointConstructor.DEFAULT_CLASSES_PATH_ON_IMAGE, true),
     // TODO: remove this once we put files in WAR into the relevant layers (i.e., dependencies,
     // snapshot dependencies, resources, and classes layers). Should copy files in the right
-    EXPLODED_WAR("exploded war", JavaEntrypointConstructor.DEFAULT_JETTY_BASE_ON_IMAGE),
-    EXTRA_FILES("extra files", "/");
+    EXPLODED_WAR("exploded war", "", true),
+    EXTRA_FILES("extra files", "/", false);
 
     private final String name;
     private final Path extractionPath;
+    private final boolean appRootRelative;
 
     /** Initializes with a name for the layer and the layer files' default extraction path root. */
-    LayerType(String name, String extractionPath) {
+    LayerType(String name, String extractionPath, boolean appRootRelative) {
       this.name = name;
       this.extractionPath = Paths.get(extractionPath);
+      this.appRootRelative = appRootRelative;
     }
 
     @VisibleForTesting
     String getName() {
       return name;
     }
 
-    @VisibleForTesting
-    Path getExtractionPath() {
+    private Path getExtractionPath() {
       return extractionPath;
     }
+
+    private boolean isAppRootRelative() {
+      return appRootRelative;
+    }
   }
 
   /** Builds with each layer's files. */
   public static class Builder {
 
     private final Map<LayerType, List<Path>> layerFilesMap = new EnumMap<>(LayerType.class);
+    private Path appRoot = Paths.get(ContainerConfiguration.DEFAULT_APP_ROOT);
```

Comment:
I'm worried path might be implementation specific (different on windows), and this should be a string? Specifically when running Jib on a windows machine.

Reply:
I had the same concern intially, but since the existing code is actually using `Path` to pass to `addEntryRecursively()`, I think it seemed better to accept `Path` for now: https://github.com/GoogleContainerTools/jib/pull/984/files#diff-1105d9e04ba077f47449d5c7095d4252L123 Maybe `addEntryRecursively()` shouldn't accept `Path`.

Reply:
Actually, your suggestion seems better. I changed it to accept Unix-style path in `String` instead of `Path`. This also removes the need for the `toUnixPath()` helper.

Reply:
Looks like addEntryRecursively actually boils down to some Unix path conversion eventually. https://github.com/GoogleContainerTools/jib/blob/8f14bd545339e372aae61522f35b783a9e2da855/jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java#L126

Reply:
Yeah, so I'm wondering all the methods and classes that need the extraction path info should get the path in `String` rather than `Path` to indicate clearly that the extraction path is and will end up in Unix-style.